crates/forge-runner/src/lib.rs

```diff
@@ -151,10 +155,21 @@ impl ValidatedForkConfig {
 impl TryFrom<RawForkParams> for ValidatedForkConfig {
     type Error = anyhow::Error;
 
-    fn try_from(value: RawForkParams) -> std::result::Result<Self, Self::Error> {
+    fn try_from(value: RawForkParams) -> Result<Self, Self::Error> {
+        let block_id = match value.block_id_type.to_lowercase().as_str() {
+            "number" => BlockId::Number(value.block_id_value.parse().unwrap()),
+            "hash" => BlockId::Hash(
+                Felt252::from(value.block_id_value.parse::<BigInt>().unwrap()).into_(),
+            ),
+            "tag" => {
+                assert_eq!(value.block_id_value, "Latest");
+                BlockId::Tag(Latest)
+            }
+            _ => unreachable!(),
```

Comment:
Ok with merging it for now but this defnitely is reachable and should be handled properly.

Reply:
It is not reachable, if you have `RawForkParams` it has to be valid

Reply:
Replaced this with `bail!` with proper error message

Reply:
#1304